--- libkawari/kawari_log.cpp	2006-09-16 00:38:48.000000000 +0800
+++ libkawari/kawari_log.cpp	2006-09-16 00:39:07.000000000 +0800
@@ -9,6 +9,7 @@
 //
 //---------------------------------------------------------------------------
 #include "libkawari/kawari_log.h"
+#include <cstdio>
 //---------------------------------------------------------------------------
 class nullstreambuf : public streambuf {
 public:
