--- src.org/gcc-mach.mak	Sat Jan 31 21:15:40 2004
+++ src/gcc-mach.mak	Sat Jan 31 22:06:04 2004
@@ -2,7 +2,7 @@
 CXX_mingw =  g++
 CXX_linux =  g++
 CXX_freebsd =g++
-CXX_darwin = c++
+CXX_darwin = g++
 
 EXEEXT_cygwin = .exe
 EXEEXT_mingw =  .exe
@@ -20,7 +20,7 @@
 DYNLIBEXT_mingw =   .dll
 DYNLIBEXT_linux =   .so
 DYNLIBEXT_freebsd = .so
-DYNLIBEXT_darwin =  .dynlib
+DYNLIBEXT_darwin =  .bundle
 
 LIBEXT_cygwin = .lib
 LIBEXT_mingw = .lib
